src/Microsoft.FileFormats/PE/PEFile.cs

```diff
@@ -105,6 +106,55 @@ public bool IsValid()
         /// </summary>
         public ImageDataDirectory ComDataDirectory { get { return ImageDataDirectory[(int)ImageDirectoryEntry.ComDescriptor]; } }
 
+        /// <summary>
+        /// Returns the address of a module export symbol if found
+        /// </summary>
+        /// <param name="symbolName">symbol name (without the module name prepended)</param>
+        /// <param name="offset">symbol offset returned</param>
+        /// <returns>true if found</returns>
+        public bool TryGetExportSymbol(string symbolName, out ulong offset)
+        {
+            try
+            {
+                ImageExportDirectory exportDirectory = _exportDirectory.Value;
+                if (exportDirectory is not null)
+                {
+                    for (int nameIndex = 0; nameIndex < exportDirectory.NumberOfNames; nameIndex++)
+                    {
+                        uint namePointerRVA = RelativeVirtualAddressReader.Read<uint>((ulong)(exportDirectory.AddressOfNames + (sizeof(uint) * nameIndex)));
+                        if (namePointerRVA != 0)
+                        {
+                            string name = RelativeVirtualAddressReader.Read<string>(namePointerRVA);
+                            if (name == symbolName)
+                            {
+                                ushort ordinalForNamedExport = RelativeVirtualAddressReader.Read<ushort>((ulong)(exportDirectory.AddressOfNameOrdinals + (sizeof(ushort) * nameIndex)));
+                                offset = RelativeVirtualAddressReader.Read<uint>((ulong)(exportDirectory.AddressOfFunctions + (sizeof(uint) * ordinalForNamedExport)));
+                                return true;
+                            }
+                        }
+                    }
+                }
+            }
+            catch (Exception ex) when (ex is InvalidVirtualAddressException || ex is BadInputFormatException)
+            {
+            }
+            offset = 0;
+            return false;
+        }
+
+        private ImageExportDirectory ReadExportDirectory()
+        {
+            if (IsValid())
+            {
+                ImageDataDirectory exportTableDirectory = ImageDataDirectory[(int)ImageDirectoryEntry.Export];
```

Comment:
Is this resilient to only this section being available? I think PEFile expects to see all of the ImageDataDirectory, which is not even true in the PE spec for a regular image. (It assumes that always 15 sections will be available, completely ignoring `SizeOfHeaders`)

Reply:
Can we assert that the Characteristics == 0?

Reply:
The symstore PE reader has always assumed that there are 15 sections and that they all are in the dump. It reads all 15 in ReadImageDataDirectory().  It sounds like you think it is important enough to only read what fits in SizeOfHeaders?  Does SizeOfHeaders include the size of the file header (ImageFileHeader) and optional header (ImageOptionalHeader)?

Should it throw an exception or do Debug.Assert if Characteristics != 0? A debug symstore isn't ever really used.

Reply:
Not sure if throw bad image, or say I can't read (return null);